app/facades/canvas_facade.rb

```diff
@@ -0,0 +1,265 @@
+      student_ids: studentIds,
+      title:       title,
+      due_at:      dueDate,
+      unlock_at:   unlockDate,
+      lock_at:     lockDate
+    })
+  end
+
+  ##
+  # Updates an existing assignment override.
+  #
+  # @param   [Integer]    courseId     the id of the course to update the override for.
+  # @param   [Integer]    assignmentId the id of the assignment to update the override for.
+  # @param   [Enumerable] studentIds   the updated student ids to provision the override to.
+  # @param   [String]     title        the updated title of the override.
+  # @param   [String]     dueDate      the updated due date for the override.
+  # @param   [String]     unlockDate   the updated date the override should unlock the assignment.
+  # @param   [String]     lockDate     the updated date the override should lock the assignment.
+  # @return  [Faraday::Response] information about the updated override.
+  def update_assignment_override(courseId, assignmentId, overrideId, studentIds, title, dueDate, unlockDate, lockDate)
+    @canvasApi.put("courses/#{courseId}/assignments/#{assignmentId}/overrides/#{overrideId}", {
+      student_ids: studentIds,
+      title:       title,
+      due_at:      dueDate,
+      unlock_at:   unlockDate,
+      lock_at:     lockDate
+  })
+  end
+
+  ##
+  # Deletes an assignment override.
+  #
+  # @param  [Integer] courseId the id of the course where the override to delete is provisioned.
+  # @param  [Integer] assignmentId the assignment for which the override to delete is provisioned.
+  # @param  [Integer] overrideId the id of the override to delete.
+  # @return [Faraday::Response] information about the deleted override.
+  def delete_assignment_override(courseId, assignmentId, overrideId)
+    @canvasApi.delete("courses/#{courseId}/assignments/#{assignmentId}/overrides/#{overrideId}")
+  end
+
+  ##
+  # Provisions a new extension to a user.
+  #
+  # @param   [Integer] courseId the course to provision the extension in.
+  # @param   [Integer] studentId the student to provisoin the extension for.
+  # @param   [Integer] assignmentId the assignment the extension should be provisioned for.
+  # @param   [String]  newDueDate the date the assignment should be due.
+  # @return  [Faraday::Response] the override that acts as the extension.
+  # @raises  [FailedPipelineError] if the creation response body could not be parsed.
+  # @raises  [NotFoundError]       if the user has an existing override that cannot be located.
+  def provision_extension(courseId, studentId, assignmentId, newDueDate)
+    overrideTitle = "#{studentId} extended to #{newDueDate}"
+    createOverrideResponse = create_assignment_override(
+      courseId,
+      assignmentId,
+      [studentId],
+      overrideTitle,
+      newDueDate,
+      get_current_formatted_time(),
+      newDueDate
+    )
+
+    # Either successful or error that is not explicitly handled here.
+    if (createOverrideResponse.status != 400)
```

Comment:
I'm a little confused by the error handling here, why is it only 400 codes that get handled by the rest of the function? Are we just assuming that any other type of error or a success is handled by the caller?